### ISSUE TITLE
Evaluate code in Debug Console during integration tests

### ISSUE THREAD

**Issue by bartekpacia** [Dart-Code/Dart-Code#4206]
### Describe the bug
**TL;DR** I can't evaluate code in Debug Console when running integration tests. I don't know if I'm doing something wrong or if this is just not possible.

Also, I can't find a way to run (using VS Code's `launch.json`) normal widget tests with `flutter run test/widget_test.dart` (=running on device) instead of `flutter test test/widget_test.dart` (running in isolation). This might be another issue, though.

### To Reproduce

**TL;DR** Run any Flutter integration test with VS Code and try to evaluate Dart code in the Debug Console.

Create `.vscode/launch.json` in the root of a Flutter project with the following contents:

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "flutter run app",
      "request": "launch",
      "type": "dart",
      "program": "lib/main.dart"
    },
    {
      "name": "flutter run integration tests",
      "request": "launch",
      "type": "dart",
      "program": "integration_test/example_test.dart"
    },
  ]
}
```

1. Clone the repository of the [Patrol framework](https://github.com/leancodepl/patrol)
1. Open the example app directory with VS Code (`$ code .`). It's located in `packages/patrol/example`
1. Now that you're in the example app's dir, put a breakpoint somewhere in the [integration_test/example_test.dart](https://github.com/leancodepl/patrol/blob/master/packages/patrol/example/integration_test/example_test.dart) file, for example on line 16:
    ```dart
    await $(FloatingActionButton).tap();
    ```
1. Run the `flutter run integration tests` configuration from `launch.json`
1. Try to evaluate some code in the Debug Console.
1. See `_compileExpression: No compilation service available; cannot evaluate from source.`.

![Screenshot 2022-10-07 at 12 55 27 AM](https://user-images.githubusercontent.com/40357511/194433168-57574402-2a6a-4e90-9f22-f4dde8086d93.png)

### Expected behavior

Being able to evaluate Dart code in Debug Console when the test is stopped on a breakpoint.

This is how it works with widget tests running in isolation:

https://user-images.githubusercontent.com/40357511/194433392-ee580ce9-764f-4cc0-8e73-c12c282c70c6.mov

### System info
 - Operating System and version: macOS 12.6 on MacBook Air M1
 - VS Code version: 1.72.0
 - Dart extension version: v3.50.0
 - Dart/Flutter SDK version: Dart 2.18.2, Flutter 3.3.3
 - Target device: Android emulator (Pixel 5 API 33)

**Comment by bartekpacia** [comment]
Thanks for answering my second little question. Works, indeed :)

But it's sad that we can't evaluate Dart code in Debug Console when running integration tests. I hope this gets implemented soon,

Thanks for your help.

**Comment by DanTup** [comment]
Yeah, I understand. Unfortunately I don't understand how the compilation service works well enough to know whether it's a big change or not. I would encourage adding 👍 's to that Flutter issue if you haven't already.

I'll close this issue as there's actually nothing to do in the VS Code extension. Once implemented in Flutter, it will just start working here.

**Comment by bartekpacia** [comment]
@DanTup Now I got this idea: is it possible to run a test file in the integration test directory with `flutter run`, not `flutter test`?

I'm thinking about something like:

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "run app",
      "request": "launch",
      "type": "dart",
      "program": "lib/main.dart"
    },
    {
      "name": "run integration test",
      "request": "launch",
      "type": "dart",
      "runTestsOnDevice": true,
      "executeWithFlutterRun": true, // <-- let's assume that this flag exists
      "program": "integration_test/example_test.dart"
    }
  ]
}

```

I haven't found a way to do the above.

In other words: when I run the `run integration test` configuration, it does `flutter test integration_test/example_test.dart`. I'd like to change it to run `flutter run -t integration_test/example_test.dart` instead. 

This would have the following benefits:
- we have the compilation service, so we're able to execute arbitrary Dart code in the Debug Console
- we can hot restart the integration test, which allows for much faster iteration and better DX

Of course, this approach isn't ideal – we'll lose the test reporting ability (and probably more `flutter test`-specific functionality.

But as a temporary workaround, I think it's an approach worth researching.

**Comment by DanTup** [comment]
@bartekpacia that's essentially what `runTestsOnDevice` does, however it only currently applies to unit tests (because integration tests already run on a device). We could perhaps add an override here, although it feels kinda clunky and much worse than being fixed in Flutter.

Let me ping on that issue and see if anyone can give an idea of the scale of work and pointers to see if it might be something I could contribute there.

**Comment by DanTup** [comment]
Re-opening this as a reminder because I think I have this working but need to come back to it.

**Comment by bartekpacia** [comment]
Thank you very much! 💙

Linking your PR: https://github.com/flutter/flutter/pull/113481

**Comment by DanTup** [comment]
I landed a fix for this in Flutter. It should be available on master now, and in some future stable Flutter SDK release.
